File: src/location/filters/TimeFilter.java
package location.filters;

import location.LocationRecord;

/**
 * Created by zsmb on 2016-07-25.
 */
public class TimeFilter extends RecordFilter {

    private long timeMS;
    private boolean acceptBefore;

    public TimeFilter(long timeMS, boolean acceptBefore) {
        this.timeMS = timeMS;
        this.acceptBefore = acceptBefore;
    }

    @Override
    protected boolean test(LocationRecord record) {
        return acceptBefore ? (record.getTimeStampMS() <= timeMS) : (record.getTimeStampMS() >= timeMS);
    }
}

File: src/main/java/location/filters/TimeFilter.kt
package location.filters


import location.LocationRecord

/**
 * Filters records before or after a given timestamp
 */
class TimeFilter(private val timeMS: Long, private val acceptBefore: Boolean) : RecordFilter() {

    override fun test(record: LocationRecord) =
            if (acceptBefore)
                record.timeStampMS <= timeMS
            else
                record.timeStampMS >= timeMS

}
